### PR TITLE
Add Checklist feature to demos

### PR DESCRIPTION
**Pull request by tatianainama**
The purpose of this PR is to add the checklist option to each of the demos.
I've created a custom hook for each of the demos, in which I listen to client's events and send them to the parent app (website).
Ideas and suggestions for new tasks are more than welcome!

### CODE REVIEW

**Comment by jaapbakker88** [comment]
LGTM @DanC5 can you have a glance as well.

File: social-messenger/src/ChecklistTasks.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { useEffect } from 'react';
+
+const notifyParent = (parent) => (message) => {
+  window.parent.postMessage(message, parent);
+}
+
+const YOUTBE_LINK = 'https://youtu.be/Ujvy-DEA-UM';
```
Comment (DanC5) [inline social-messenger/src/ChecklistTasks.js:7]: little typo here :)

**Comment by DanC5** [comment]
This looks great, just a little typo of `YOUTBE` in some places. Though do we need to add this to our local env files `REACT_APP_TARGET_ORIGIN`?

**Review by DanC5 — APPROVED** [review]

**Comment by tatianainama** [comment]
> Though do we need to add this to our local env files `REACT_APP_TARGET_ORIGIN`?

It wouldn't be necessary, only if you want to run the `website` and the `demo` locally (and test the communication between them).

**Review by jaapbakker88 — APPROVED** [review]

File: customer-support/src/ChecklistTasks.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          break;
+        default:
+          break;
+      }
+    }
+    if (customerClient) {
+      customerClient.on(handleNewEvent);
+    }
+    return () => customerClient?.off(handleNewEvent);
+  })
```
Comment (vishalnarkhede) [inline customer-support/src/ChecklistTasks.js:28]: No hook dependency?

Comment (vishalnarkhede) [inline customer-support/src/ChecklistTasks.js:28]:
```suggestion
  useEffect(() => {
    const handleNewEvent = ({ type }) => {
      switch(type) {
        case 'message.new':
          notify(SEND_MESSAGE);
          break;
        default:
          break;
      }
    }
    if (customerClient) {
      customerClient.on(handleNewEvent);
    }
    return () => customerClient?.off(handleNewEvent);
  }, [customerClient])
```

Comment (tatianainama) [reply customer-support/src/ChecklistTasks.js]: Thanks! Added in [438a9db](https://github.com/GetStream/website-react-examples/pull/7/commits/438a9dbb14bdee6073bb4d1ea9d185df4ca594b8)

**Review by vishalnarkhede — APPROVED** [review]

**Pull request merged**